### PR TITLE
Refactored same code in ya_intern.c

### PR DESCRIPTION
**Pull request by nmiculinic**
DRY -- don't repeat yourself. Much code was repeated,
mostly error control. Also used ferror instead of checking
return code from fscanf. Resulting code is cleaner and shorter.

### CODE REVIEW

File: src/intern_blks/ya_intern.c [hunk 1]
```diff
@@ -225,37 +232,10 @@ void ya_int_bandwidth(ya_block_t * blk) {
...
-		//Close if one of the files is opened
-		if(rxfile)
-			fclose(rxfile);
-		if(txfile)
-			fclose(txfile);
-		pthread_detach(blk->thread);
-		pthread_exit(NULL);
-	}
-	else {
-		if(fscanf(rxfile, "%lu", &orx)!=1)
```
Comment (NBonaparte) [inline src/intern_blks/ya_intern.c]: You should get `orx` and `otx` here.

File: src/intern_blks/ya_intern.c [hunk 2]
```diff
@@ -394,23 +347,10 @@ void ya_int_diskio(ya_block_t *blk) {
...
-	tfile = fopen(tpath, "r");
-	if (tfile == NULL) {
-		fprintf(stderr, "Error opening file %s\n", tpath);
-		strncpy(blk->buf, "BLOCK ERROR!", strlen("BLOCK ERROR!"));
-		ya_draw_pango_text(blk);
-		pthread_detach(blk->thread);
-		pthread_exit(NULL);
-	}
-	else {
-		if(fscanf(tfile,"%lu %lu %lu %lu %lu %lu %lu %lu %lu %lu %lu", &tdo[0], &tdo[1], &tdo[2], &tdo[3], &tdo[4], &tdo[5], &tdo[6], &tdo[7], &tdo[8], &tdo[9], &tdo[10])!=11)
```
Comment (NBonaparte) [inline src/intern_blks/ya_intern.c]: Again, scanning for `tdo` should be done here.

File: src/intern_blks/ya_intern.c [hunk 3]
```diff
@@ -343,23 +313,8 @@ void ya_int_cpu(ya_block_t *blk) {
...
 		space = 0;
-	tfile = fopen(fpath, "r");
-	if (tfile == NULL) {
-		fprintf(stderr, "Error opening file (%s)\n", fpath);
-		strncpy(blk->buf, "BLOCK ERROR!", strlen("BLOCK ERROR!"));
-		ya_draw_pango_text(blk);
-		pthread_exit(NULL);
-	}
-	else {
-		if(fscanf(tfile,"%s %Lf %Lf %Lf %Lf",cpustr, &old[0],&old[1],&old[2],&old[3])!=5)
```
Comment (NBonaparte) [inline src/intern_blks/ya_intern.c]: Same here; the `old` values should be obtained before the loop.

**Comment by nmiculinic** [comment]
To be honest, I'm not using yabar anymore and I don't remember what changes I made a year ago. 

If you want to use some of the code you're more than welcome, but I have other priorities right now.

**Comment by NBonaparte** [comment]
No worries, thanks for the reply.

**Pull request merged**